core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -158,6 +163,26 @@ public KafkaSchemaRegistry(SchemaRegistryConfig config,
     this.masterNodeSensor.add(m, new Gauge());
   }
 
+  /**
+   * A Schema Registry instance's identity is in part the port it listens on. Currently the
+   * port can either be configured via the deprecated `port` configuration, or via the `listeners`
+   * configuration.
+   *
+   * This method uses `Application.parseListeners()` from `rest-utils` to get a list of listeners, and
+   * returns the port of the first listener to be used for the instance's identity.
+   *
+   * In theory, any port from any listener would be sufficient. Choosing the first, instead of say the last,
+   * is arbitrary.
+   */
+  // TODO: once RestConfig.PORT_CONFIG is deprecated, remove the port parameter.
+  static int getPortForIdentity(int port, List<String> configuredListeners) {
+    List<URI> listeners = Application.parseListeners(configuredListeners, port,
+            Arrays.asList(SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_SSL,
+                    SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_PLAINTEXT),
+            SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_PLAINTEXT);
+    return listeners.get(0).getPort();
```

Comment:
@ewencp: I pushed a new commit that introduces docs explaining this behavior. I also created a new issue that addresses the issue you discovered, with a workaround: https://github.com/confluentinc/schema-registry/issues/386